### PR TITLE
Remove compatibility mode

### PR DESCRIPTION
**Pull request by joshuahuburn**
#### Description
Small PR to remove old and dead code which is no longer needed with v15.
Note this is into the dev branch and so not affecting master.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 1]
```diff
@@ -521,7 +513,7 @@ const FString& FSpatialLaunchConfigDescription::GetDefaultTemplateForRuntimeVari
 {
 	switch (GetDefault<USpatialGDKEditorSettings>()->GetSpatialOSRuntimeVariant())
 	{
-	case ESpatialOSRuntimeVariant::CompatibilityMode:
+	case ESpatialOSRuntimeVariant::CompatibilityMode_DEPRECATED:
 		if (GetDefault<USpatialGDKSettings>()->IsRunningInChina())
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp:517]: Ah should probably remove these.

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 2]
```diff
@@ -227,7 +227,7 @@ namespace ESpatialOSRuntimeVariant
 enum Type
 {
 	Standard,
-	CompatibilityMode
+	CompatibilityMode_DEPRECATED
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h:230]: Why would we not just remove it?

**Review by mironec — APPROVED** [review]
Approving, as it looks good. Left just one question

File: Setup.bat [hunk 3]
```diff
@@ -124,7 +123,6 @@ if defined DOWNLOAD_MOBILE (
     spatial package retrieve worker_sdk      c-dynamic-x86_64-clang_ndk21-android      %PINNED_CORE_SDK_VERSION%   %DOMAIN_ENVIRONMENT_VAR%   "%CORE_SDK_DIR%\worker_sdk\c-dynamic-x86_64-clang_clang_ndk21-android.zip"
 )
     spatial package retrieve worker_sdk      csharp                                     %PINNED_CORE_SDK_VERSION%   %DOMAIN_ENVIRONMENT_VAR%   "%CORE_SDK_DIR%\worker_sdk\csharp.zip"
-    spatial package retrieve spot            spot-win64                                 %PINNED_SPOT_VERSION%       %DOMAIN_ENVIRONMENT_VAR%   "%BINARIES_DIR%\Programs\spot.exe"
```
Comment (jessicafalk) [inline Setup.bat:127]: also do it for Setup.sh. or will this be done later?

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h (hunk 2 above)
Comment (jessicafalk) [reply SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h:230]: +1

**Review by jessicafalk — APPROVED** [review]

File: Setup.bat (hunk 3 above)
Comment (joshuahuburn) [reply Setup.bat:127]: Good point.

**Pull request merged**